src/Client/Grpc/GrpcDurableTaskClient.cs

```diff
@@ -321,18 +325,20 @@ public override AsyncPageable<OrchestrationMetadata> GetAllInstancesAsync(Orches
 
     /// <inheritdoc/>
     public override Task<PurgeResult> PurgeInstanceAsync(
-        string instanceId, CancellationToken cancellation = default)
+        string instanceId, PurgeInstanceOptions? options = null, CancellationToken cancellation = default)
     {
+        bool recursive = options?.Recursive ?? true;
```

Comment:
default to `false` to avoid breaking behavior change. Please also address this in all other locations for purge and terminate.

If we want the behavior to be `true` by default we will need to do that in a major version rev. You can add a TODO comment for that in the options class.

Reply:
Done